openghg/store/_infer_time.py

```diff
@@ -9,7 +9,7 @@
 logger = logging.getLogger("openghg.store")
 logger.setLevel(logging.DEBUG)  # Have to set level for logger as well as handler
 
-TupleTimeType = Tuple[Union[int, float], str]
+TupleTimeType = Tuple[Union[int, float, None], Union[str, None]]
```

Comment:
Maybe this should be a dataclass with a classmethod that does what `parse_period` does?

Reply:
Could you add that as a suggestion to Issue #911 ?

Reply:
Okay, I've done that. The comment here is more around readability: `TupleTimeType` is somewhat complicated and the parts have a clear meaning. You could also just use a named tuple, e.g. `TimePeriod = namedtuple("TimePeriod",  ["value", "unit"])`, although maybe the `NamedTuple` class from `typing` would be better:

```
from typing import NamedTuple

class TimePeriod(NamedTuple):
   value: Union[int, float, None] = None
   unit: Optional[str] = None

input_freq  = parse_period(period) if period else TimePeriod()
```

Reply:
This doesn't really affect the use of `parse_period` since `tp = TimePeriod(1, "H")` and `u, v = tp` give `u == 1` and `v == "H"`.

Reply:
Also, `TimeTupleType` has different definitions in `util._time` and `store._infer_time`, so they should be synchronised. I guess I would suggest defining a `typing.NamedTuple` subclass to replace `TupleTimeType` in `util._time` and importing that in `store._infer_time`, although importing the type alias could work.

Reply:
Ok - I've integrated in the suggested `TimePeriod` class into `openghg.types._types` for this and used this in the two places this is defined. This can be revisited when addressing Issue #911 if needed.